src/core/lib/security/credentials/external/aws_request_signer.cc

```diff
@@ -94,15 +94,14 @@ AwsRequestSigner::AwsRequestSigner(
     static_request_date_ =
         absl::FormatTime(kXAmzDateFormat, request_date, absl::UTCTimeZone());
   }
-  url_ = grpc_uri_parse(url, false);
-  if (url_ == nullptr) {
+  absl::StatusOr<URI> tmp_url = URI::Parse(url);
```

Comment:
@renkelvin Please take a look at the changes in this file and let me know if you have any questions.